addon/adapters/osf-adapter.js

```diff
@@ -11,13 +12,55 @@ export default DS.JSONAPIAdapter.extend(DataAdapterMixin, {
     host: config.OSF.apiUrl,
     namespace: config.OSF.apiNamespace,
 
-    buildURL(modelName, id, snapshot, requestType, query) {  // jshint ignore:line
-        var url = this._super(...arguments);
-        // Fix issue where CORS request failed on 301s: Ember does not seem to append trailing
+    buildURL(modelName, id, snapshot, requestType, query, dirtyRelationship) { // jshint ignore:line
+	      var url;
+        var links = dirtyRelationship ? snapshot.record.get(
+            `links.relationships.${Ember.String.underscore(dirtyRelationship)}.links`
+        ) : false;
+        if (links) {
+            url = links.self ? links.self.href : links.related.href;
+        } else {
+            url = this._super(...arguments);
+        }
+	// Fix issue where CORS request failed on 301s: Ember does not seem to append trailing
         //  slash to URLs for single documents, but DRF redirects to force a trailing slash
-        if (url.lastIndexOf('/') !== 0) {
+        if (url.lastIndexOf('/') !== url.length - 1) {
             url += '/';
         }
         return url;
+    },
+    relationshipPayload(snapshot, dirty, store) {
+        var relationMeta = snapshot.record[dirty].meta();
+        var relationType = relationMeta.type;
+        var serialized;
+        if (relationMeta.options.serializer) {
+          serialized = relationMeta.options.serializer(snapshot);
+        } else {
+          var serializer = store.serializerFor(relationType);
+          var toBeSent = snapshot.record.get(dirty).filter(
+            function(record){
+              return record.id === null;
+            }
+          );
+          //under the assumption relationship saves are atomic, only one component is added at a time.
```

Comment:
This has a different implication. We want the assumption that only one related field can be changed at a time not that only one record per field can change at a time.

For the institutions case, for example, we to update the Node with all of the added institutions. Once https://openscience.atlassian.net/browse/EOSF-42, this behavior will also work for contributors/components.